src/app/trade/page.tsx

```
@@ -0,0 +1,909 @@
+    hash: approvalHash 
+  });
+
+
+  // Auto-retry trade after successful approval
+  useEffect(() => {
+    if (isApprovalConfirmed && approvalHash) {
+      // Clear the approval hash and execute the trade
+      setApprovalHash(undefined);
+      executeTrade();
+    }
+  }, [isApprovalConfirmed, approvalHash,executeTrade]);
```

Biome (2.1.2)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)